src/snowflake/snowpark/dataframe.py

```diff
@@ -1374,6 +1376,10 @@ def select(
                         f"Called '{table_func.user_visible_name}' and '{e.user_visible_name}'."
                     )
                 table_func = e
+                if _emit_ast and ast:
```

Comment:
```suggestion
                if ast:
```
_emit_ast must be True if ast is not None.  The other branches are inconsistent here as well.

Reply:
This may be confusing, we sometimes use in the code `ast=None` to auto-fill the ast (e.g., in the context of Column/Literals). Therefore we need both `_emit_ast` and `ast` in the check. I agree this is not great, we should clean it up and simplify.

Reply:
Will not take the (valid) suggestion yet. We have cleanup to do around the logic that enables/disables AST creation.